server/src/main/java/org/opensearch/search/query/TopDocsCollectorContext.java

```diff
@@ -183,7 +183,9 @@ CollectorManager<?, ReduceableSearchResult> createManager(CollectorManager<?, Re
                 );
             } else {
                 if (hitCount == -1) {
-                    if (trackTotalHitsUpTo != SearchContext.TRACK_TOTAL_HITS_ACCURATE) {
+                    if (trackTotalHitsUpTo == SearchContext.TRACK_TOTAL_HITS_ACCURATE) {
```

Comment:
This is my bad, missed that sadly :( thanks @jed326 !

Reply:
This condition is when `track_total_hits=true` and there is `terminate_after` as well in the request. For non-concurrent case, that seem to return the relation always with `Eq` in response (See [here](https://github.com/opensearch-project/OpenSearch/pull/10082/files#diff-b8c66f0b315e7a5e88a41e8567ac0fe2d32c93175758e397078334dff4724a1cR156)). Whereas for non-concurrent case I think it will return different relation in the response (either `gte` or `eq`) depending on if request was early terminated (code ref [here](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/search/query/TotalHitCountCollectorManager.java#L75)). So we will need to handle this case by checking if track_total_hits is set to accurate or not. This is similar to the case 2 you are calling out [here](https://github.com/opensearch-project/OpenSearch/issues/8371#issuecomment-1726761922) but for size 0.

Reply:
@sohami agreed that we should fix this behavior. I'm already planning on taking this up in the follow-up to https://github.com/opensearch-project/OpenSearch/issues/8371 (and will include a lot more testing in that change), so why don't we take up the `size=0` case in that follow-up as well?

Reply:
fine with that. thanks!